contracts/UsulVetoGuard.sol

```diff
@@ -0,0 +1,147 @@
+        vetoVoting = IVetoVoting(_vetoVoting);
+        ozLinearVoting = IOZLinearVoting(_ozLinearVoting);
+        usul = IUsul(_usul);
+
+        emit UsulVetoGuardSetup(
+            msg.sender,
+            _owner,
+            _vetoVoting,
+            _ozLinearVoting,
+            _usul
+        );
+    }
+
+    /// @notice Queues a transaction for execution
+    /// @param proposalId The ID of the proposal to queue
+    function queueProposal(uint256 proposalId) external {
+        // If proposal is not yet timelocked, then finalize the strategy
+        if(usul.state(proposalId) == 0) ozLinearVoting.finalizeStrategy(proposalId);
+        
+        require(usul.state(proposalId) == 2, "Proposal must be timelocked before queuing");
+
+        // While loop is used since the Usul interface does not support getting the quantity of TX hashes
+        // stored within a given proposal. This loops through and gets the hash from each index until the call
+        // reverts, and then the function is exited
+        uint256 txIndex;
+        while (true) {
+            try usul.getTxHash(proposalId, txIndex) returns (bytes32 txHash) {
+                transactionQueuedBlock[txHash] = block.number;
+                txIndex++;
+            } catch {
+                require(txIndex > 0, "Invalid proposal ID");
+
+                emit ProposalQueued(msg.sender, proposalId);
+
+                return;
+            }
+        }
+    }
+
+    /// @notice This function is called by the Gnosis Safe to check if the transaction should be able to be executed
+    /// @notice Reverts if this transaction cannot be executed
+    /// @param to Destination address.
+    /// @param value Ether value.
+    /// @param data Data payload.
+    /// @param operation Operation type.
+    /// @param safeTxGas Gas that should be used for the safe transaction.
+    /// @param baseGas Gas costs for that are independent of the transaction execution(e.g. base transaction fee, signature check, payment of the refund)
+    /// @param gasPrice Maximum gas price that should be used for this transaction.
+    /// @param gasToken Token address (or 0 if ETH) that is used for the payment.
+    /// @param refundReceiver Address of receiver of gas payment (or 0 if tx.origin).
+    function checkTransaction(
+        address to,
+        uint256 value,
+        bytes memory data,
+        Enum.Operation operation,
+        uint256 safeTxGas,
+        uint256 baseGas,
+        uint256 gasPrice,
+        address gasToken,
+        address payable refundReceiver,
+        bytes memory,
+        address
+    ) external view override {
+        bytes32 txHash = usul.getTransactionHash(to, value, data, operation);
+
+        require(
+            transactionQueuedBlock[txHash] > 0,
+            "Transaction has not been queued yet"
+        );
+
+        require(!vetoVoting.getIsVetoed(txHash), "Transaction has been vetoed");
+
+        require(!vetoVoting.isFrozen(), "DAO is frozen");
```

Comment:
@mudrila and here it's ensuring the DAO is not frozen or the proposal isn't vetoed